### DISCUSSION TITLE
Default highlight language for code blocks

### DISCUSSION THREAD

**Discussion by Kanaduchi** [squidfunk/mkdocs-material#3417] [Q&A] [answered]
Hi!

Currently I have more than 100 code blocks with yaml syntax. And in each code block I have to specify yaml language:
`` yaml

Is it possible to set yaml language as default language for whole project? I don't like to put it in each code block

**Comment by facelessuser** [comment, marked as answer]
There is no easy way as fenced code blocks just aren't specced to work like you are suggesting, but there is a way to do so with [custom fences](https://facelessuser.github.io/pymdown-extensions/extensions/superfences/#custom-fences). You'd have to override the default fences and wrap its logic with some that specify the language as `yaml` if no language is provided.

I'm on vacation right now, so I don't really have time to show an example here, but when I get back, I can probably illustrate what I mean.
